python-lib/api_call.py

```diff
@@ -153,7 +157,19 @@ def query(url: str, headers: dict, parameters: dict) -> dict:
     :returns: API's response
     :rtype: dict
     """
-    response = requests.get(url=url, headers=headers, params=parameters)
+    successful_get = False
+    attempt_number = 0
+    while not successful_get and attempt_number <= Constants.MAX_RETRIES:
+        try:
+            attempt_number += 1
+            response = requests.get(url=url, headers=headers, params=parameters)
+            successful_get = True
+        except Exception as err:
+            logger.warning("ERROR:{}".format(err))
+            logger.warning("on attempt #{}".format(attempt_number))
+            if attempt_number == Constants.MAX_RETRIES:
+                raise LinkedinPluginError("Error in batch processing on attempt #{}: {}".format(attempt_number, err))
```

Comment:
Maybe we could return a slightly more user-friendly message such as : "The request that you sent to LinkedIn took longer than expected despite {attempt_number} attempts. Consider decreasing the number of account ids or checking on your internet connection."

Reply:
Maybe, it could be nice to specify which dataset is causing the issue

Reply:
Following your comment, a suggestion to reduce the number of ids/batch size has been added in case of an error 400.